### PR TITLE
feat: Allow localhost connection strings

### PR DESCRIPTION
**Pull request by elliott-with-the-longest-name-on-github**
If a connection string is on `localhost`, don't check for whether it's pooled or non-pooled. Just allow it.

### CODE REVIEW

**Review by adriancooney — APPROVED** [review]
Looks good!

File: packages/postgres/src/postgres-connection-string.ts [hunk 1]
```diff
@@ -37,3 +37,29 @@ export function isPooledConnectionString(connectionString: string): boolean {
 export function isDirectConnectionString(connectionString: string): boolean {
   return !isPooledConnectionString(connectionString);
 }
+
+export function isLocalhostConnectionString(connectionString: string): boolean {
+  try {
+    // This seems silly, but we can use all of the hard work put into URL parsing
+    // if we just convert `postgresql://` to `https://` and then parse it as a URL.
+    const withHttpsProtocol = connectionString.startsWith('postgresql://')
```
Comment (klaussner) [inline packages/postgres/src/postgres-connection-string.ts:45]: I think this should work without replacing the protocol before parsing the URL:
```
> new URL("postgresql://localhost:5432")
URL {
  href: 'postgresql://localhost:5432',
  origin: 'null',
  protocol: 'postgresql:',
  username: '',
  password: '',
  host: 'localhost:5432',
  hostname: 'localhost',
  port: '5432',
  pathname: '',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

Comment (elliott-with-the-longest-name-on-github) [reply packages/postgres/src/postgres-connection-string.ts:45]: It does not in some cases 😢 Try running the tests without it.

Comment (elliott-with-the-longest-name-on-github) [reply packages/postgres/src/postgres-connection-string.ts:45]: image

Actually, this very example doesn't work

Comment (klaussner) [reply packages/postgres/src/postgres-connection-string.ts:45]: TIL that [most browsers don’t follow the WHATWG URL spec](https://bugzilla.mozilla.org/show_bug.cgi?id=1374505) when it comes to parsing URLs with unknown protocols ([example](https://felixfbecker.github.io/whatwg-url-custom-host-repro/)). Parsing `postgresql:` connection strings with Node.js should work, though (the tests are all passing). 🤔

**Pull request merged**